packages/calypso-e2e/src/lib/components/sidebar-component.ts

```diff
@@ -35,22 +39,81 @@ export class SidebarComponent extends BaseContainer {
 	 */
 	async _postInit(): Promise< void > {
 		await this.page.waitForLoadState( 'networkidle' );
+		this.sidebar = await this.page.waitForSelector( selectors.sidebar );
 	}
 
 	/**
-	 * Clicks on the sidebar menu item matching the name.
-	 * Note that the menu item must be visible in some shape or form.
-	 * If there are multiple elements that match the selector, the first
-	 * matching element will be clicked (as per Playwright documentation).
-	 * Waits and assertions should be placed on the page/component that this
-	 * method will cause a navigation to, to ensure readiness before the script
-	 * continues.
+	 * Given heading and subheading, or any combination of the two, locate and click on the items on the sidebar.
 	 *
-	 * @param {string} name Plaintext name of the menu item in the sidebar.
+	 * This method supports any of the following use cases:
+	 *   - heading only
+	 *   - subheading only
+	 *   - heading and subheading
+	 *
+	 * Heading is defined as the top-level menu item that is permanently visible on the sidebar, unless outside
+	 * of the viewport.
+	 *
+	 * Subheading is defined as the child-level menu item that is exposed only on hover or main heading being toggled.
+	 *
+	 * Note, in the current Nav Unification paradigm, clicking on certain combinations of sidebar menu items will cause
+	 * a navigation away to an entirely new page (eg. wp-admin). Attempting to reuse the SidebarComponent object
+	 * under this condition will trigger an exception.
+	 *
+	 * @param {{[key: string]: string}} param0 Named object parameter.
+	 * @param {string} param0.heading Plaintext representation of the top level heading.
+	 * @param {string} param0.subheading Plaintext representation of the child level heading.
 	 * @returns {Promise<void>} No return value.
 	 */
-	async clickMenuItem( name: string ): Promise< void > {
-		await this.page.click( `text=${ toTitleCase( name ) }` );
+	async gotoMenu( {
+		heading,
+		subheading,
+	}: {
+		heading: string;
+		subheading?: string;
+	} ): Promise< void > {
+		if ( heading ) {
+			await this._click( { selector: `${ selectors.menuText } >> text=${ heading.trim() }` } );
```

Comment:
Just a suggestion for a case-insensitive match:
```suggestion
			await this._click( { selector: `${ selectors.menuText } :text-matches("${ heading.trim() }", "i")` } );
```

Reply:
If I use `toTitleCase` method here that should address the case issue, correct?

As a bonus the `toTitleCase` can also work for the subheading.